lib/visualization/draw_event_stream.py

```diff
@@ -207,30 +208,64 @@ def plot_events(xs, ys, ts, ps, save_path=None, num_compress='auto', num_show=10
 
+
     #Plot images
-    if len(imgs)>0 and show_frames:
-        for imgidx, (img, img_ts) in enumerate(zip(imgs, img_ts)):
-            img = img[crop[0]:crop[1], crop[2]:crop[3]]
-            if len(img.shape)==2:
-                img = np.stack((img, img, img), axis=2)
-            if num_compress > 0:
-                events_img = events_to_image(xs[0:num_compress], ys[0:num_compress],
-                        np.ones(num_compress), sensor_size=img.shape[0:2])
-                events_img[events_img>0] = 1
-                img[:,:,1]+=events_img[:,:]
-                img = np.clip(img, 0, 1)
-            x, y = np.ogrid[0:img.shape[0], 0:img.shape[1]]
-            event_idx = np.searchsorted(ts, img_ts)
-
-            ax.scatter(xs[0:event_idx], ts[0:event_idx], ys[0:event_idx], zdir='z',
-                    c=colors[0:event_idx], facecolors=colors[0:event_idx],
-                    s=np.ones(xs.shape)*event_size, marker=marker, linewidths=0, alpha=1.0 if show_events else 0)
-
-            ax.plot_surface(y, img_ts, x, rstride=stride, cstride=stride, facecolors=img, alpha=1)
-
-            ax.scatter(xs[event_idx:-1], ts[event_idx:-1], ys[event_idx:-1], zdir='z',
-                    c=colors[event_idx:-1], facecolors=colors[event_idx:-1],
-                    s=np.ones(xs.shape)*event_size, marker=marker, linewidths=0, alpha=1.0 if show_events else 0)
+    if show_frames:
+        if len(imgs) > 0:    # gray scale frame
+            for imgidx, (img, img_ts_) in enumerate(zip(imgs, img_ts)):
+                img = img[crop[0]:crop[1], crop[2]:crop[3]]
+                if len(img.shape)==2:
+                    img = np.stack((img, img, img), axis=2)
+                if num_compress > 0:
+                    events_img = events_to_image(xs[0:num_compress], ys[0:num_compress],
+                            np.ones(num_compress), sensor_size=img.shape[0:2])
+                    events_img[events_img>0] = 1
+                    img[:,:,1]+=events_img[:,:]
+                    img = np.clip(img, 0, 1)
+                x, y = np.ogrid[0:img.shape[0], 0:img.shape[1]]
+                event_idx = np.searchsorted(ts, img_ts_)
+                if event_idx > 0:
+                    ax.scatter(xs[0:event_idx], ts[0:event_idx], ys[0:event_idx], zdir='z',
+                            c=colors[0:event_idx], facecolors=colors[0:event_idx],
+                            s=np.ones(xs.shape)[0:event_idx]*event_size, marker=marker, linewidths=0, alpha=1.0 if show_events else 0)
+
+                ax.plot_surface(y, img_ts_, x, rstride=stride, cstride=stride, facecolors=img, alpha=1)
+
+                ax.scatter(xs[event_idx:-1], ts[event_idx:-1], ys[event_idx:-1], zdir='z',
+                        c=colors[event_idx:-1], facecolors=colors[event_idx:-1],
+                        s=np.ones(xs.shape)[event_idx:-1]*event_size, marker=marker, linewidths=0, alpha=1.0 if show_events else 0)
+        else:   # event image
```

Comment:
this is our modification. when `show_frames == True`  but `len(imgs) == 0` i.e. no `imgs` are provided as argument, it create event image.